.github/workflows/ci-validate-platforms.yml

```diff
@@ -3,224 +3,109 @@ name: Validate Platforms/Browsers
-      env:
-        PLAYWRIGHT_BROWSER: chromium
-
-    - name: Test Playwright (firefox)
-      run: yarn workspace @microsoft/fast-components test-pw
-      continue-on-error: true
-      timeout-minutes: 2
-      env:
-        PLAYWRIGHT_BROWSER: firefox
-
-    - name: Test Playwright (webkit)
-      run: yarn workspace @microsoft/fast-components test-pw
-      continue-on-error: true
-      timeout-minutes: 2
-      env:
-        PLAYWRIGHT_BROWSER: webkit
+      - name: Set git to use LF
+        if: ${{ matrix.os == 'windows-latest' }}
+        run: |
+          git config --global core.autocrlf false
+          git config --global core.eol lf
+
+      - name: Checkout Branch
+        uses: actions/checkout@v2
+
+      - if: ${{ github.event_name == 'pull_request' }}
+        run: |
+          git fetch --no-tags --prune --depth=1 origin +refs/heads/master:refs/remotes/origin/master
+
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo '::set-output name=dir::$(yarn cache dir)'
+
+      - name: Set up node_modules cache
+        uses: actions/cache@v2
+        id: yarn-cache
+        with:
+          path: |
+            ${{ steps.yarn-cache-dir-path.outputs.dir }}
+            node_modules
+            */*/node_modules
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+
+      - name: Install package dependencies
+        run: yarn install --frozen-lockfile --ignore-scripts --network-timeout=36000
+
+      - name: Test Prettier format
+        run: yarn format:check
+
+      - name: Prepare workspaces
+        run: yarn prepare
+
+      - name: Run tests in changed packages
+        if: ${{ github.event_name == 'pull_request' }}
+        run: yarn lerna run test --stream --since=origin/master
+
+      - name: Run tests in all Lerna Packages
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/master' }}
+        run: yarn lerna run test --stream
+
+      - name: Install playwright dependencies and browsers
+        run: |
+          yarn playwright install-deps
+          yarn playwright install
+
+      - name: Build Playwright tests
+        run: yarn workspace @microsoft/fast-components build:pw
+
+      - name: Test Playwright (chromium)
+        run: yarn workspace @microsoft/fast-components test-pw
+        continue-on-error: true
+        timeout-minutes: 2
+        env:
+          PLAYWRIGHT_BROWSER: chromium
+
+      - name: Test Playwright (firefox)
+        run: yarn workspace @microsoft/fast-components test-pw
+        continue-on-error: true
+        timeout-minutes: 2
+        env:
+          PLAYWRIGHT_BROWSER: firefox
+
+      - name: Test Playwright (webkit)
+        run: yarn workspace @microsoft/fast-components test-pw
+        continue-on-error: true
+        timeout-minutes: 2
+        env:
+          PLAYWRIGHT_BROWSER: webkit
+
+      - name: Initialize CodeQL
```

Comment:
CodeQL and Analysis could also be broken out into a security test job.  Breaking into separate jobs may require an upload to cloud of the built bits and then download into those jobs so you don't have to rerun build after the initial time.

Reply:
Ideally we only build once, and all other jobs would use those build artifacts. I'll make a follow-up issue.